src/offchain-ipfs-callables.md

```diff
@@ -1,14 +1,175 @@
+
+### `ipfs_cat_bytes`
+
+Displays the bytes behind a given CID
+
+**Frequency:** Every other block<br />
+**Weight**: 100,000
+
+#### Arguments
+
+- `cid`: A CID string, e.g. `QmU1f6ngsoHvwtViihzLQPXCA8j3sagmvY9GJJDY7Ao7Aa`
+
+#### Returns
+
+The requested bytes - UTF-8 as a string, non-UTF-8 as hexidecimal string
+
+### `ipfs_connect`
+
+Connects the embedded node to the given Multiaddr.
+
+**Frequency:** Every block<br />
+**Weight**: 100,000
+
+#### Arguments
+
+- `multiaddr`: A string, e.g. `/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`
+
+#### Returns
+
+???
+
+### `ipfs_disconnect`
+
+Disconnects from the given Multiaddr.
+
+**Frequency:** Every block<br />
+**Weight**: 500,000
+
+#### Arguments
+
+- `multiaddr`: A string, e.g. `/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`
+
+#### Returns
+
+???
+
+### `ipfs_dht_findpeer`
+
+Performs a search for the addresses associated with the provided PeerId.
+
+**Frequency:** Every block<br />
+**Weight**: 100,000
+
+#### Arguments
+
+- `peerID`: A `PeerId` string, e.g. `QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ`
+
+#### Returns
+
+An "opaque" multiaddr, i.e. one without a `PeerId`, such as `/ip4/104.131.131.82/tcp/4001`.
+
+### `ipfs_dht_findproviders`
+
+Search for PeerIds known to be providing the given Cid. You must connected to at least one peer!
+
+**Frequency:** Every block<br />
+**Weight**: 100,000
+
+#### Arguments
+
+- `cid`: The CID string of your desired content, e.g. `QmY7Yh4UquoXHLPFo2XbhXkhBvFoPwmQUSa92pxnxjQuPU`
+
+#### Returns
+
+- `peerID`: An array `PeerId` string, e.g. `[QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ]`
+
+### `ipfs_insert_pin`
+
+Non-recursively pins a block with the specified Cid, protecting it from removal.
+
+**Frequency:** Every other block<br />
+**Weight**: 100,000
+
+#### Arguments
+
+- `cid`: The CID string data you want to pin, e.g. `QmU1f6ngsoHvwtViihzLQPXCA8j3sagmvY9GJJDY7Ao7Aa`
+
+#### Returns
+
+???
+
+### `ipfs_remove_pin`
+
+Removes a pin from a block, so that it is no longer persistent and can be removed.
+
+**Frequency:** Every other block<br />
+**Weight**: 100,000
+
+#### Arguments
+
+- `cid`: The CID string data you want to unpin, e.g. `QmU1f6ngsoHvwtViihzLQPXCA8j3sagmvY9GJJDY7Ao7Aa`
+
+#### Returns
+
+### `ipfs_remove_block`
+
+Removes a block from the node’s repository
+
+**Frequency:** Every other block<br />
+**Weight**: 300,000
+
+#### Arguments
+
+- `cid`: The CID string data you want to unpin, e.g. `QmU1f6ngsoHvwtViihzLQPXCA8j3sagmvY9GJJDY7Ao7Aa`
+
+#### Returns
+
+???
+
+## Functionality _not_ exposed in the example pallet
+
+`offchain::ipfs` provides access to the full feature set of Rust IPFS, of which the callables above
+are only a subset. In addition, the following APIs can possibly be exposed:
+
+- Publishing pubsub messages
+- Reporting bitswap Stats
+- Identity and local addresses
+- Listing out local references
```

Comment:
Are these accurate?

Reply:
there's plenty more that we don't expose from the IPFS node; I don't think we need to list them - these only have partial implementations available, which is in line with our note about how this is not the whole IPFS functionality or the only right way of exposing it

Reply:
Removed the whole section